### PR TITLE
[coq_lib] Enforce that boot libraries depend on no theories

### PR DESCRIPTION
**Pull request by ejgallego**
Signed-off-by: Emilio Jesus Gallego Arias <e+git@x80.org>

### CODE REVIEW

File: src/dune_rules/coq_stanza.ml [hunk 1]
```diff
@@ -145,6 +149,8 @@ module Theory = struct
        and+ modules = Stanza_common.modules_field "modules"
        and+ enabled_if = Enabled_if.decode ~allowed_vars:Any ~since:None ()
        and+ buildable = Buildable.decode in
+       (* boot libraries cannot depend on other theories *)
+       if boot && List.is_non_empty buildable.theories then boot_has_deps buildable;
```
Comment (rgrinberg) [inline src/dune_rules/coq_stanza.ml:153]: The location of `List.hd buildable.theories` should be a bit more useful to the user in this error.

**Comment by ejgallego** [comment]
I was thinking of enforcing the invariant in the stanza datatype with a `Boot_or_theories.t` type, but IMHO it seems overkill.

**Comment by rgrinberg** [comment]
> I was thinking of enforcing the invariant in the stanza datatype with a `Boot_or_theories.t` type, but IMHO it seems overkill.

Indeed it seems like wasted effort if we just make the stdlib a first class library.

File: src/dune_rules/coq_stanza.ml (hunk 1 above)
Comment (ejgallego) [reply src/dune_rules/coq_stanza.ml]: Done thanks.

**Review by rgrinberg — APPROVED** [review]

**Comment by rgrinberg** [comment]
@ejgallego could you push a CHANGES entry for this?

**Comment by ejgallego** [comment]
Yes, I didn't do a CHANGES as `(boot)` is really "internal" so far, but I will signal it.

**Comment by ejgallego** [comment]
Note that indeed `(boot)` is (purposely) undocumented.

**Pull request merged**